src/core/handshaker/handshaker.cc

```diff
@@ -104,6 +104,16 @@ void HandshakeManager::DoHandshake(
   args_.args = channel_args;
   args_.event_engine = args_.args.GetObject<EventEngine>();
   args_.acceptor = acceptor;
+  // Add a channelz trace that we're performing a handshake.
+  // Note that we only commit this to the log if we see an error - otherwise
```

Comment:
I may be missing something here, since the template logic is really hard to follow, but doesn't each call to GRPC_CHANNELZ_LOG() wind up calling Commit()?